### PR TITLE
ISPN-6779 Improve Security Configuration screen

### PR DESCRIPTION
**Pull request by vblagoje**
Master only. Do not integrate yet. Feedback and review, please @ryanemerson

### CODE REVIEW

File: src/main/webapp/components/directives/cache/security/security.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            scope.data.roles.push(role);
+          }
+        };
+
+        scope.isRoleSelected = function (role) {
+          return scope.data.roles.indexOf(role) > -1;
+        };
+
+        scope.isAnyFieldModified = function () {
+          return true;
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: Should this always be true?

**Comment by ryanemerson** [comment]
@vblagoje A few thoughts:

1. As we are now using it's own directive, and such features are possible, should we disable the selecting of roles if security is not currently enabled?

2. As we provide an 'undo change' operation throughout the cache config screens, should we not do the same here? My thoughts are that we should ideally allow this for the role selectors, but at the very least it should be there for the 'enable' checkbox.

**Comment by vblagoje** [comment]
@ryanemerson done on both points. Please review again, squash and integrate!

File: src/main/webapp/components/directives/cache/security/security.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.security', ['ispn.services.utils']);
+
+  module.directive('security', ['utils', '$modal', function (utils, modal) {
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: $modal is redundant

File: src/main/webapp/components/directives/cache/security/security.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          scope.internalController = {};
+        }
+        scope.hasSecurityBeenDefinedGlobally = function () {
+          return utils.isNotNullOrUndefined(scope.cluster.security) && utils.isNotNullOrUndefined(scope.cluster.security.SECURITY.authorization.AUTHORIZATION) ;
+        };
+
+        if (utils.isNullOrUndefined(scope.data)) {
+          scope.data = {};
+          utils.deepSet(scope.data, "SECURITY.authorization.AUTHORIZATION", {});
+          scope.data = scope.data.SECURITY.authorization.AUTHORIZATION;
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: This does not work as expected when we pass the data through to the `cache-creation-controller.service` as we have not set the 'is-new-node' field to true on either `security.SECURITY` or `security.SECURITY.authorization.AUTHORIZATION` nodes.  This results in the following error when we attempt to update an existing config with security.SECURITY undefined:

```WFLYCTL0216: Management resource '[ ("profile" => "clustered"), ("subsystem" => "datagrid-infinispan"), ("cache-container" => "clustered"), ("configurations" => "CONFIGURATIONS"), ("distributed-cache-configuration" => "default"), ("security" => "SECURITY") ]' not found```

File: src/main/webapp/components/directives/cache/security/security.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          scope.metadata[field].uiModified = false;
+          scope.metadata[field].style = null;
+        };
+
+        scope.fieldChangeRequiresRestart = function (field) {
+          return utils.isNotNullOrUndefined(scope.metadata[field]) && scope.metadata[field]['restart-required'] !== 'no-services';
+        };
+
+        scope.requiresRestart = function () {
+          return false;
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: Should this be false? The resource description states `"restart-required" => "resource-services"` for both the enabled and roles attributes.

File: src/main/webapp/components/directives/cache/security/security.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        };
+
+        scope.undoFieldChange = function (field) {
+          scope.data[field] = scope.prevData[field];
+          scope.metadata[field].uiModified = false;
+          scope.metadata[field].style = null;
+        };
+
+        scope.fieldChangeRequiresRestart = function (field) {
+          return utils.isNotNullOrUndefined(scope.metadata[field]) && scope.metadata[field]['restart-required'] !== 'no-services';
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: Currently this is not utilised anywhere, but I think it should be based upon my comments below. The template will also need updating to reflect this.

File: src/main/webapp/components/directives/cache/security/security.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+  module.directive('security', ['utils', '$modal', function (utils, modal) {
+    return {
+      restrict: 'E',
+      scope: {
+        data: '=',
+        cluster: '=',
+        metadata: '=',
+        initDefaults: '=',
+        readOnly: '=',
```
Comment (ryanemerson) [inline src/main/webapp/components/directives/cache/security/security.js]: Minor, but we don't actually use `initDefaults` or `readOnly` at present. Maybe we should utilise readOnly for the checkboxes ng-disabled field? Currently I don't think initDefaults is applicable as we only have checkbox inputs in this directive.

**Comment by ryanemerson** [comment]
@vblagoje Integrated ed88e77bd9db08acd5d5791e670f90177d2404c6, thanks.

**Pull request closed without merging**